docs/root/faq/performance/how_to_benchmark_envoy.rst

```diff
@@ -0,0 +1,41 @@
+What are best practices for benchmarking Envoy?
+===============================================
+
+There is :ref:`no single QPS, latency or throughput overhead <faq_how_fast_is_envoy>` that can
+characterize a network proxy such as Envoy. Instead, any measurements need to be contextually aware,
+ensuring an apples-to-apples comparison with other systems by configuring and load testing Envoy
+appropriately. As a result, we can't provide a canonical benchmark configuration, but instead offer
+the following guidance:
+
+* A release Envoy binary should be used. If building, please ensure that `-c opt`
+  is used on the Bazel command line. When consuming Envoy point releases, make
+  sure you are using the latest point release; given the pace of Envoy development
+  it's not reasonable to pick older versions when making a statement about Envoy
+  performance. Similarly, if working on a master build, please perform due diligence
+  and ensure no regressions or performance improvements have landed proximal to your
+  benchmark work and that your are close to HEAD.
+
+* The :option:`--concurrency` Envoy CLI flag should be unset (providing one worker thread per
+  logical core on your machine) or set to match the number of cores/threads made available to other
+  network proxies in your comparison.
+
+* Disable :ref:`generate_request_id
+  <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.generate_request_id>`.
+
+* Disable :ref:`dynamic_stats
+  <envoy_v3_api_field_extensions.filters.http.router.v3.Router.dynamic_stats>`.
+
+* Ensure that the networking and HTTP filter chains are reflective of comparable features
+  in the systems that Envoy is being compared with.
```

Comment:
When benchmarking the overhead of a proxy compared to a direct connection, I wonder if it might make sense to disable *all* stats (are there stats that are needed for functional correctness???). The reason is that direct connections would provide no stats.

Or at least highlight that a key value-add for Envoy is observability of all paths, and that's not free. But if we benchmark with stats disabled, which can be done with a stat-matcher config, we might get more of an apples-to-apples comparison, and then be able to add in the broken-out cost of the stats.

Reply:
Agree, updated.